storagedriver/factory/factory.go

```diff
@@ -58,7 +58,12 @@ func Create(name string, parameters map[string]interface{}) (storagedriver.Stora
 		// }
 		// return driverClient, nil
 	}
-	return driverFactory.Create(parameters)
+	d, err := driverFactory.Create(parameters)
```

Comment:
![](http://t0.gstatic.com/images?q=tbn:ANd9GcQNLbaVviqeMUHmw7wyV35a8x2XXbPI75UzZX0P6Iq541IZ2dgsyQ)

Text-book one-size-fits-all truths like "Thou shalt not write redundant code" are certainly fine as generic guidelines. Though, experience proved following them blindly is generally a bad sign of over-engineering.

We need to stay practical here, and work well as a "team" with varied focus and interests - hence to accept that, sometimes, things are not exactly like we wished they were:-)

About this PR:

 * what's the expected benefit? (religious belief A or B does *not* count as "benefit")
 * are we still on-track to get that benefit?
 * what changes does @stevvooe expect in order for this to be mergeable?
 * are any of these changes radically altering the answer to the first two questions?
  * yes? then indeed there is a need to discuss it
  * no? then let's implement these changes and have this merged

> Final call is yours, that's not my code

Things don't work like that here.
This is *our* code.
Neither yours or mine.

At least, that's the spirit that I would like everyone to follow.

Hope that helps.